### ISSUE TITLE
Report the total error for simulations with data

### ISSUE THREAD

**Issue by teutonicod** [Open-Systems-Pharmacology/PK-Sim#1471]
Hi,
I would like to propose a feature request, if it is feasible :-).
When building a model, in particular if the parameter are adjusted manually, we have very often the issue of comparing different models and/or different parametrization. In order to do that more quantitatively, it would be very useful to have the total error between the simulation and the data. 

What I am thinking about is basically what is created at the beginning of a parameter optimization, where the current simulation output (with the initial parameters) is compared to the data. 

As difference, I understand that in a parameter identification, there is a phase where the output is mapped to the data, but in a simulation an automatic mapping could be done as it is done at the beginning of the PI. I am not saying that the user should be able to change this mapping, since I think this would need major changes, but simply to have the total error based on the automatic mapping, and communicating to the user, which output is mapped to which data series. In my experience, this automatic mapping works very very often. 
Please let me know what do you think!

Donato

**Comment by StephanSchaller** [comment]
I like the idea!

Maybe this could be a feature of "Comparison Plots"? Both for Individual and for population simulations.

**Comment by msevestre** [comment]
I think we could enable a lot of features if we could manage to define somewhere what obs data matches what output. Just having it work very often is good, but not enough. The definition could be done once and reuse everywhere else.

This is for example required for https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/1056

where we want to graph predicted vs observed.

**Comment by teutonicod** [comment]
Could this be done at the simulation level, in a tab like the one "Chart editor"?

**Comment by msevestre** [comment]
I think this does not belong at the simulation level. This is some knowledge that is more at the project level.

**Comment by msevestre** [comment]
Obviosuly it could be created from a simulation (probably where it makes sense because one can select output), but the definition of such a mapping is almost like a building block, or an extension of the observed data set. This would allow a lot of automation

**Comment by teutonicod** [comment]
No I don't think so. In several cases, we can compare data to plasma profiles in one simulation and to blood profiles in another, or simulate a specific organ and mach the observations. This will depend on the specific simulation and not at the project level. 
So I would imagine much better that the user create the simulation where an output is generated, then drag the observations in the simulation and for instance an observed vs pred plot is generated based on the default mapping (as in the PI toolbox). If it is fine, nothing to do, but if there is any problem, the user could modify the mapping in the tab of the simulation.

I think that on an higher level it is done at the project level, since if the data are imported correctly, there is a mapping to PK-Sim output during the import. I would see the option at the simulation level as the possibility to modify this mapping if needed.

**Comment by teutonicod** [comment]
Well, unless we re-imagine the observation building block and do the mapping at that level, as it is already done up to a certain level

**Comment by msevestre** [comment]
Are you saying you compare the same SET of observed data to different outputs?  🤔
Because you don't know what was measured? 

I don't think this makes sense to define it at the simulation level. For the case where you don't know what the data is about, then you can override the stuff maybe in the simulation .But in general, you want to compare plasma concentration with plasma concentration and not with brain tissue

**Comment by teutonicod** [comment]
No, I use the a specific set of data always to the same output (at least in most of the cases), but the output can be different in different simulations.
A mapping plasma <-> plasma or brain<->brain, isn't done already when importing the data?

**Comment by msevestre** [comment]
> isn't done already when importing the data?

It is in PK-Sim but not in MoBi. Maybe this is enough at the observed data level. I seemed to remember that we had discussed this Mapping with @Yuri05 and we agreed that we needed something else

@Yuri05 Do you remember?

**Comment by msevestre** [comment]
After thinking about this with @teutonicod yesterday, I think this could work like this

1. During import of observed data, user specifies organ and compartment (as it is now)
1. In the simulation there is a new tab were specific obs data <-> output mapping can be defined. 
Very similar to what we have in the PI
1. By default, the mapping would be the one inferred from step 1. However this could be overwritten on a simulation basis
1. When dragging new obs data onto a simulation, we DO NOT add a new line in this mapping. Only user specific stuff is added so that changes in obs data are reflected in all simulation directly
1. When using a simulation for PI, this information is used first to crate the auto mapping and default to the global stuff if not available

This would allow also to create the new plots in #1056

We would have to add the same type of functionality in MoBi to have the same plot available as well => implementation will be in core

Thoughts?

**Comment by PavelBal** [comment]
Sound great, would also make creations of PI easier.

**Comment by StephanSchaller** [comment]
Great stuff!

**Comment by Yuri05** [comment]
@teutonicod I think all of that is implemented now in 11.1 - please check the release notes of the **beta**-release
https://github.com/Open-Systems-Pharmacology/Forum/discussions/1394

**Comment by Yuri05** [comment]
@teutonicod Please close the issue if you think it's implemented in 11.1 (I think it is :))
